### PR TITLE
Give bots ability to auto-replace harvesters

### LINKED ISSUE #15962: Make HarvesterManagerBotModule request harv replacements
After #15922 is merged, it will be rather trivial to make `HarvesterManagerBotModule` request harvesters being built if their number is below the number of refineries, making the AI a bit better at keeping/restoring their economy.

### PR DESCRIPTION
**Pull request by reaperrr**
...when their number drops below the number of refineries.

~Depends on #15982.~

Closes #15962.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/BotModules/HarvesterBotModule.cs [hunk 1]
```diff
@@ -99,6 +111,30 @@ void IBotTick.BotTick(IBot bot)
...
+				}
+			}
+		}
+
+		bool ShouldBuildHarvester()
+		{
+			if (!Info.HarvesterTypes.Any())
+				return false;
+
+			if (AIUtils.CountBuildingByCommonName(Info.HarvesterFactoryTypes, player) == 0)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/BotModules/HarvesterBotModule.cs]: If you check the `unitBuilder.RequestedProductionCount` here instead then we should be able to drop the factory types completely - it'll queue a single harv, then wait until that is completed before doing anything else.  This simplifies the yaml config, and also saves perf.

Comment (reaperrr) [reply OpenRA.Mods.Common/Traits/BotModules/HarvesterBotModule.cs]: Done and moved the whole check inline into BotTick.

**Comment by reaperrr** [comment]
Updated. moved the 'should build' decision making inline into BotTick, and added a BotDebug message for external unit build requests in the unit builder module.

Since the pt hasn't been tagged yet, maybe we can still sneak this in?

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/168534c9b3fe99dbda6c59e8fd836cbd9a25c9ec...141bb6f8f911e108b9125fa4de656ae42314b0ed)

**Pull request merged**